src/api/mod.rs

```diff
@@ -1,6 +1,31 @@
 mod handler;
 mod schema;
 mod server;
+pub mod tap;
 
 pub use schema::build_schema;
 pub use server::Server;
+pub use tap::{TapControl, TapRegister};
+
+use tokio::sync::mpsc;
+
+// Control tx/rx channels use Tokio unbounded channels to take advantage of async on the
+// receiver end, and sync on the sender. This allows us to trigger a control message when
+// the returned stream is canceled externally (e.g. the subscription terminating on the client)
+// by implementing `Drop`.
+
+pub type ControlSender = mpsc::UnboundedSender<ControlMessage>;
+pub type ControlReceiver = mpsc::UnboundedReceiver<ControlMessage>;
+
+/// Control messages that can be sent from GraphQL requests that affect the operation
+/// of topology.
+pub enum ControlMessage {
+    Tap(tap::TapControl),
+}
```

Comment:
I initially tried augmenting `signal::SignalTo`, and found:

1. The existing `#[derive(Debug, Clone, Copy, Eq, PartialEq)]` isn't conducive to receiving a `tap::TapControl`. This could potentially be reworked, but again, with the mandate of not wanting to bend things solely to accommodate tap, I punted.

2. Signals are (currently) specific to OS-level interrupts. This didn't feel like the best place to reference an `api::TapSink`.

3. There may be other requirements for the API to interface with topology in the future, such as the API providing a GraphQL mutation for updating configuration ad-hoc. It felt like a `api::ControlMessage` enum provided a better home for that than top-level OS signals.